fsae-vehicle-fw/src/callbacks.cpp

```diff
@@ -0,0 +1,51 @@
+// Anteater Electric Racing, 2025
+
+#include <ADC.h>
+#include "callbacks.h"
+#include "peripherals/adc.h"
+
+#define LOGIC_LEVEL_V 3.3
+#define ADC1_MAX_VALUE adc->adc1->getMaxValue()
+
+void StartADCPinReadings() {
+    uint8_t currentPin = pins[adcIndex];
+    Serial.print( "Starting read for ");
+    Serial.println(currentPin);
+    Serial.print("time since prev start of read");
+    uint32_t currentTime = micros();
+    Serial.println(currentTime - begin);
+    begin = micros();
+    adc->adc1->startSingleRead(currentPin); // in callbacks.h
+
+}
+
+void ADCReadingCompleteCallback () {
+    // read from pin
+    // store pin reads
+    // go to next pin
+
+    uint16_t sensorRead = adc->adc1->readSingle();
+    adcReads[adcIndex] = sensorRead*LOGIC_LEVEL_V/ADC1_MAX_VALUE; // Changing to this formula for now to see if it helps w apps fault sensorRead;
+    Serial.print( "In ADCReadingCompleteCallback pin is ");
+    Serial.println(pins[adcIndex]);
+    Serial.print( "In ADCReadingCompleteCallback sensorRead is ");
+    Serial.println(sensorRead);
+    Serial.print( "In ADCReadingCompleteCallback value is ");
+    Serial.println(sensorRead*3.3/adc->adc0->getMaxValue(), DEC);
+
+    // increment to next sensor pin & start read
+    ++adcIndex;
+    if(adcIndex == SENSOR_PIN_AMT){ // Do here so we don't start a read for an invalid pin
+        adcIndex = 0;
+        Serial.print("end of loop thinks value of begin is ");
+        Serial.println(begin);
+        Serial.print("duration of all reads: ");
+        uint32_t end = micros();
+        Serial.println(end - begin);
+        return;
+    }
+    uint16_t currentPin = pins[adcIndex];
+    adc->adc1->enableInterrupts(ADCReadingCompleteCallback);
+    bool successfulStart = adc->adc1->startSingleRead(currentPin); // in callbacks.h
```

Comment:
```suggestion
    adc->adc1->startSingleRead(currentPin); // in callbacks.h
```
could we change it to this when removing debugging code?

Reply:
removed